src/components/templates/PartyMap/PartyMap.tsx

```diff
@@ -12,74 +12,183 @@ import { useSelector } from "hooks/useSelector";
-            return acc;
-          }, {})
-        : {},
-    [usersInCamp]
-  );
+  const selectRoom = useCallback((room: PartyMapRoomData) => {
+    setSelectedRoom(room);
+    setRoomModalOpen(true);
+  }, []);
 
-  const modalHidden = useCallback(() => {
-    setIsRoomModalOpen(false);
+  const unselectRoom = useCallback(() => {
+    // TODO: should we rework this so we only unselect if it matches? To prevent race conditions?
+    // setRoomClicked((prevRoomClicked) =>
+    //   prevRoomClicked === room.title ? undefined : room.title
+    // );
+    setSelectedRoom(undefined);
+    setRoomModalOpen(false);
   }, []);
 
+  // TODO: can we get rid of this in favour of unselectRoom?
+  const closeRoomModal = useCallback(() => {
+    setRoomModalOpen(false);
+  }, []);
+
+  // Find current room from url
   const { roomTitle } = useParams();
+  const currentRoom = useMemo(() => {
+    if (!currentVenue || !roomTitle) return;
+
+    return currentVenue.rooms.find(
+      (venueRoom) =>
+        createUrlSafeName(venueRoom.title) === createUrlSafeName(roomTitle)
+    );
+  }, [currentVenue, roomTitle]);
 
   useEffect(() => {
-    if (roomTitle) {
-      const partyRoom = venue?.rooms.find(
-        (room) => createUrlSafeName(room.title) === createUrlSafeName(roomTitle)
-      );
-      if (partyRoom) {
-        setSelectedRoom(partyRoom);
-        setIsRoomModalOpen(true);
-      }
+    if (currentRoom) {
+      selectRoom(currentRoom);
     }
-  }, [roomTitle, setIsRoomModalOpen, setSelectedRoom, venue]);
+  }, [currentRoom, selectRoom]);
+
+  // TODO: do we need/want to calculate this on the frontend? Or can we do it in a function/similar serverside?
+  const usersInVenue = useCampPartygoers(currentVenue.name);
+
+  // TODO: Need a better name for this.. What is it actually doing?
+  const attendances = useMemo(() => {
+    if (!usersInVenue) return {};
+
+    // TODO: is this counting logic correct?
+    return usersInVenue.reduce<Record<string, number>>((acc, value) => {
+      Object.keys(value.lastSeenIn).forEach((lastSeenInKey) => {
+        acc[lastSeenInKey] = (acc[lastSeenInKey] ?? 0) + 1;
+      });
+
+      return acc;
+    }, {});
+  }, [usersInVenue]);
+
+  // TODO: do we need this?
+  // const [showEventSchedule, setShowEventSchedule] = useState(false);
 
   return (
     <>
       <div className="party-venue-container">
+        <AnnouncementMessage message={currentVenue.bannerMessage} />
+
+        {/* TODO: should this still be here on the partymap? */}
+        {/*{usersInCamp && (*/}
+        {/*  <div className="row">*/}
+        {/*    <div className="col">*/}
+        {/*      <UserList*/}
+        {/*        users={usersInCamp}*/}
+        {/*        imageSize={50}*/}
+        {/*        disableSeeAll={false}*/}
+        {/*        isCamp={true}*/}
+        {/*        activity={currentVenue.activity ?? "partying"}*/}
+        {/*      />*/}
+        {/*    </div>*/}
+        {/*  </div>*/}
+        {/*)}*/}
```

Comment:
Chris says: now that you can click around on the map, may not need to be part of it.

Though not everyone will necessarily click on the actual map.

Actually.. maybe we should keep it. Even if it's an option we can turn on/off. Also need to design/review with Ed as to whether we can fulfill this purpose in a better way.

Reply:
Update: it would be better to keep the `UserList` but default it to off. We can demo with it on and see how the clients react, then use that to decide if it stays off.

Longer term we should design something that serves the same purpose - showing off who's at the party. It also gives the party a sense of liveliness, fullness.

Reply:
backlogged in https://github.com/sparkletown/internal-sparkle-issues/issues/134

Reply:
cleaned up in https://github.com/sparkletown/sparkle/pull/840/commits/65c7bb71e136693f9cf501cc03f6b9c434034f87